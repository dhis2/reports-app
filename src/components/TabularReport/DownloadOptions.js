import React from 'react'
import PropTypes from 'prop-types'
import i18n from '../../utils/i18n/locales'
import styles from '../../utils/styles'

export const DownloadOptions = props => (
    <div id="download-options-container" style={styles.downloadContainer}>
        <span
            style={styles.downloadButton}
            role="button"
            tabIndex="0"
            onClick={props.onDownloadXlsClick}
        >
            {i18n.t('download as xls')}
        </span>
    </div>
)

DownloadOptions.propTypes = {
    onDownloadXlsClick: PropTypes.func.isRequired,
}
