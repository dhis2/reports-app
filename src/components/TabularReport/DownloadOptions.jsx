import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'
import React from 'react'
import styles from './DownloadOptions.module.css'

const prefix = i18n.t('Download as')

export const DownloadOption = ({ file }) => (
    <a
        className={styles.downloadButton}
        href={file.url}
        download
        tabIndex="0"
        rel="noreferrer"
        target={file.extension === 'pdf' ? '_blank' : '_self'}
    >
        {prefix} {file.extension}
    </a>
)

DownloadOption.propTypes = {
    file: PropTypes.shape({
        extension: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired,
    }),
}

export const DownloadOptions = ({ fileUrls }) => (
    <div className={styles.downloadOptionsContainer}>
        {fileUrls.map((file) => (
            <DownloadOption file={file} key={file.url} />
        ))}
    </div>
)

DownloadOptions.propTypes = {
    fileUrls: PropTypes.array.isRequired,
}
