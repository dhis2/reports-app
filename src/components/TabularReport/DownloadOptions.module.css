.downloadButton {
    cursor: pointer;
    outline: none;
    color: #757575;
    font-size: 12px;
    text-transform: uppercase;
    margin-left: 8px;
}

.downloadOptionsContainer {
    display: flex;
    justify-content: flex-end;
}
