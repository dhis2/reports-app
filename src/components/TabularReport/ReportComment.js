import React from 'react'
import PropTypes from 'prop-types'
import { InputField, Button } from '@dhis2/d2-ui-core'
import i18n from '@dhis2/d2-i18n'

const isCommentingActionEnabled = comment => comment.trim()
const actionButtonPlaceholder = i18n.t('Share')
const inputFieldPlaceholder = i18n.t(
    'Write a comment, question or interpretation of this report'
)

export const ReportComment = props => (
    <div id="share-component">
        <div>
            <InputField
                placeholder={inputFieldPlaceholder}
                type="text"
                multiline
                fullWidth
                value={props.comment}
                onChange={props.setDataSetReportComment}
            />
            <Button
                raised
                color="primary"
                onClick={props.shareDataSetReportComment}
                disabled={!isCommentingActionEnabled(props.comment)}
            >
                {actionButtonPlaceholder}
            </Button>
        </div>
        <style jsx>{`
            div > div {
                margin: 16px 0;
            }
        `}</style>
    </div>
)

ReportComment.propTypes = {
    comment: PropTypes.string.isRequired,
    setDataSetReportComment: PropTypes.func.isRequired,
    shareDataSetReportComment: PropTypes.func.isRequired,
}
