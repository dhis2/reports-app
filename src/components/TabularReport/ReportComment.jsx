import i18n from '@dhis2/d2-i18n'
import { InputField, Button } from '@dhis2/d2-ui-core'
import PropTypes from 'prop-types'
import styles from './ReportComment.module.css'

const isCommentingActionEnabled = (comment) => comment.trim()
const actionButtonPlaceholder = i18n.t('Share')
const inputFieldPlaceholder = i18n.t(
    'Write a comment, question or interpretation of this report'
)

export const ReportComment = (props) => (
    <div id="share-component">
        <div className={styles.wrapper}>
            <InputField
                placeholder={inputFieldPlaceholder}
                type="text"
                multiline
                fullWidth
                value={props.comment}
                onChange={props.setDataSetReportComment}
            />
            <Button
                raised
                color="primary"
                onClick={props.shareDataSetReportComment}
                disabled={!isCommentingActionEnabled(props.comment)}
            >
                {actionButtonPlaceholder}
            </Button>
        </div>
    </div>
)

ReportComment.propTypes = {
    comment: PropTypes.string.isRequired,
    setDataSetReportComment: PropTypes.func.isRequired,
    shareDataSetReportComment: PropTypes.func.isRequired,
}
