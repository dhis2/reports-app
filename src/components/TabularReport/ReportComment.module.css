.wrapper {
    margin: 16px 0;
}
