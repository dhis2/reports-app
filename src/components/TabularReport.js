import React from 'react'
import PropTypes from 'prop-types'
import { DownloadOptions } from './TabularReport/DownloadOptions'
import ReportTable from './TabularReport/ReportTable'
import ReportLoader from './ReportLoader'
import './TabularReport/styles.css'

const TabularReport = ({ content, isLoading, fileUrls }) => (
    <ReportLoader content={content} isLoading={isLoading}>
        <div className="tabular-report">
            <DownloadOptions fileUrls={fileUrls} />
            <ReportTable content={content} />
        </div>
    </ReportLoader>
)

TabularReport.propTypes = {
    content: PropTypes.object.isRequired,
    isLoading: PropTypes.bool.isRequired,
    fileUrls: PropTypes.array.isRequired,
}

export default TabularReport
