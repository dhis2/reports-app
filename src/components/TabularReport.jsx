import PropTypes from 'prop-types'
import React from 'react'
import ReportLoader from './ReportLoader.jsx'
import { DownloadOptions } from './TabularReport/DownloadOptions.jsx'
import ReportTable from './TabularReport/ReportTable.jsx'

const TabularReport = ({ content, isLoading, fileUrls }) => (
    <ReportLoader content={content} isLoading={isLoading}>
        <div className="tabular-report">
            <DownloadOptions fileUrls={fileUrls} />
            <ReportTable content={content} />
        </div>
    </ReportLoader>
)

TabularReport.propTypes = {
    content: PropTypes.object.isRequired,
    fileUrls: PropTypes.array.isRequired,
    isLoading: PropTypes.bool.isRequired,
}

export default TabularReport
