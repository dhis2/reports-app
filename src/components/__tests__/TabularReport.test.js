/* eslint-disable */
/* React */
import React from 'react'

/* unit testing tools */
import { shallow } from 'enzyme'

import TabularReport from '../TabularReport'

const content = {
    title: 'Bo - Child Health - Jan to Mar 2019',
    headers: [
        'Name',
        'Child Health Actual reports',
        'Child Health Expected reports',
        'Child Health Reporting rate',
        'Child Health Actual reports on time',
        'Child Health Reporting rate on time',
    ],
    rows: [
        ['Bo', '290.0', '378.0', '76.7', '290.0', '76.7'],
        ['Gbo', '3.0', '6.0', '50.0', '3.0', '50.0'],
    ],
}
const fileUrls = [
    {
        extension: 'xls',
        url: 'xls_url',
    },
    {
        extension: 'csv',
        url: 'csv_url',
    },
]

describe('<TabularReport/>', () => {
    it('Renders a loader when loading is true', () => {
        const wrapper = shallow(
            <TabularReport content={{}} isLoading={true} fileUrls={[]} />
        )
        expect(wrapper).toMatchSnapshot()
    })

    it('Renders nothing when content is an empty object and loading is false', () => {
        const wrapper = shallow(
            <TabularReport content={{}} isLoading={false} fileUrls={[]} />
        )
        expect(wrapper.html()).toBeNull()
    })
    it('Renders a table when loading is false and there is content to display', () => {
        const wrapper = shallow(
            <TabularReport
                content={content}
                isLoading={false}
                fileUrls={fileUrls}
            />
        )
        expect(wrapper).toMatchSnapshot()
    })
})
