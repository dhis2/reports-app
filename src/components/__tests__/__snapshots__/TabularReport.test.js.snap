// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabularReport/> Renders a loader when loading is true 1`] = `
<div
  className="tabular-report__loader"
>
  <CircularProgress
    large={false}
    small={false}
    style={Object {}}
  />
</div>
`;

exports[`<TabularReport/> Renders a table when loading is false and there is content to display 1`] = `
<div
  className="tabular-report"
>
  <Component
    fileUrls={
      Array [
        Object {
          "extension": "xls",
          "url": "xls_url",
        },
        Object {
          "extension": "csv",
          "url": "csv_url",
        },
      ]
    }
  />
  <ReportTable
    content={
      Object {
        "headers": Array [
          "Name",
          "Child Health Actual reports",
          "Child Health Expected reports",
          "Child Health Reporting rate",
          "Child Health Actual reports on time",
          "Child Health Reporting rate on time",
        ],
        "rows": Array [
          Array [
            "Bo",
            "290.0",
            "378.0",
            "76.7",
            "290.0",
            "76.7",
          ],
          Array [
            "Gbo",
            "3.0",
            "6.0",
            "50.0",
            "3.0",
            "50.0",
          ],
        ],
        "title": "Bo - Child Health - Jan to Mar 2019",
      }
    }
  />
</div>
`;
